### PR TITLE
Allow moving saved models

### PR DESCRIPTION
**Pull request by lilleswing**
Add arg to save to specify saving to a filepath.

Can be helpful if playing around in a jupyter notebook and deciding to save a model after construction.

### CODE REVIEW

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 1]
```diff
@@ -885,6 +897,8 @@ def save(self):
     self.rnn_final_states = rnn_final_states
     self.rnn_zero_states = rnn_zero_states
     self.session = session
+    if save_dir is not None:
+      shutil.copytree(self.model_dir, save_dir)
```
Comment (peastman) [inline deepchem/models/tensorgraph/tensor_graph.py]: Should it also set `self.model_dir = save_dir`?

Comment (lilleswing) [reply deepchem/models/tensorgraph/tensor_graph.py]: I think of when you pass "save_dir" to this arg it is creating a "snapshot" of the model at that point in time.  

If you change self.model_dir = save_dir then if you "fit" again the save_dir version will change, not the current model_dir. 

I think the behavior shown here is the more intuitive behavior.

Comment (rbharath) [reply deepchem/models/tensorgraph/tensor_graph.py]: Agree with @lilleswing that not changing `self.model_dir` is more intuitive

Comment (peastman) [reply deepchem/models/tensorgraph/tensor_graph.py]: Make sure to clearly document that behavior then.  It's the opposite of what I would expect.

**Comment by rbharath** [comment]
I think that since we're now using Keras' model saving capabilities this PR is no longer relevant. I'm going to close the PR, but let me know if we should re-open.

**Pull request closed without merging**